setup.py

```diff
@@ -19,23 +42,64 @@
         "tests/data/*.txt",
         "*.yaml",
     ],
+
     "jwst.fits_generator": [
         "templates/*.inc",
         "templates/*.txt",
         "tests/okfile/*.prop",
     ],
+
     "jwst.lib": ["tests/data/siaf.db"],
+
     # Include the rules .py files in associations test data
     "jwst.associations": ["tests/data/*.py"],
+
+    # Include C extensions
+    "jwst.lib.src": ["*.c"],
+
     # Include the transforms schemas
     "jwst.transforms": ["schemas/stsci.edu/jwst_pipeline/*.yaml"],
     "jwst.stpipe.resources": ["schemas/*.yaml"],
 }
 
+# Install packages required for this setup to proceed:
+_install_setup_requires(dict(setup_requires=SETUP_REQUIRES))
+for dep_pkg in SETUP_REQUIRES:
+    try:
+        importlib.import_module(dep_pkg)
+    except ImportError:
+        print(f"{dep_pkg} is required in order to install '{PACKAGENAME}'.\n"
+              f"Please install {dep_pkg} first.",
+              file=sys.stderr)
+        exit(1)
+
+# Setup C module include directories
+import numpy
+include_dirs = [numpy.get_include()]
+
+# Setup C module macros
+define_macros = [('NUMPY', '1')]
+
+# Handle MSVC `wcsset` redefinition
+if sys.platform == 'win32':
+    define_macros += [
+        ('_CRT_SECURE_NO_WARNING', None),
+        ('__STDC__', 1)
+    ]
+
 setup(
     use_scm_version=True,
-    setup_requires=["setuptools_scm"],
+    setup_requires=SETUP_REQUIRES,
+    install_requires=INSTALL_REQUIRES,
```

Comment:
These are not necessary (nor any of the code above defining the global variables), as pyproject.toml makes sure that `numpy` is installed before `setup()` is run.  And the meta already gets pulled from setup.cfg, as specified in pyproject.toml.

Reply:
But if this is not incorrect, i.e., it does not hurt, I would prefer to keep this code in.

As a test, I created a new/clean Python environment and I commented out the code that you say is optional and tried to install `jwst` into this environment. Here is the error that I get:

```python
(clnenv) dacia:jwst mcara$ pip install .
Processing /.../jwst
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /.../clnenv/bin/python3 /.../clnenv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /var/folders/kb/78_n024d2jb8p24bf6t2c3_c0001_0/T/tmpbnsu8zik
       cwd: /private/var/folders/kb/78_n024d2jb8p24bf6t2c3_c0001_0/T/pip-req-build-ul60c3p9
  Complete output (16 lines):
  Traceback (most recent call last):
    File "/.../clnenv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 280, in <module>
      main()
    File "/.../clnenv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/.../clnenv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 114, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/private/var/folders/kb/78_n024d2jb8p24bf6t2c3_c0001_0/T/pip-build-env-5c_84cv7/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 154, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "/private/var/folders/kb/78_n024d2jb8p24bf6t2c3_c0001_0/T/pip-build-env-5c_84cv7/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 135, in _get_build_requires
      self.run_setup()
    File "/private/var/folders/kb/78_n024d2jb8p24bf6t2c3_c0001_0/T/pip-build-env-5c_84cv7/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 150, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 77, in <module>
      import numpy
  ModuleNotFoundError: No module named 'numpy'
  ----------------------------------------
```

Should users be able to install `jwst` by clonning it and then running `pip install .`?

Reply:
Yes, if you have numpy listed in `pyproject.toml`, then it _will_ work.  This is the modern way to do it.

Reply:
I.e. the only reason you are getting to the point of not having `numpy` in the above traceback is because it is not listed in pyproject.toml as it needs to be.  `jwst` is PEP517-compliant, and this is how one guarantees that the build requirements are installed.

Reply:
@jdavies-st Alright. I moved `'numpy'` requirement to `pyproject.toml` and now `pip install .` works but the now `python setup.py install` fails:

```python
jwst mcara$ python setup.py install
WARNING: The wheel package is not available.
Traceback (most recent call last):
  File "setup.py", line 78, in <module>
    import numpy
ModuleNotFoundError: No module named 'numpy'
```

Reply:
You need `import numpy` at the top of `setup.py`.  pyproject.toml will guarantee it is installed in the isolated build environment, but you'll still need to import it.

Be sure to use `oldest-supported-numpy` in pyproject.toml, so that we avoid the problems we saw a few months ago when the numpy folks delevered 1.20 and we were still using 1.19 in our deliveries to SDP and it broke the installs.